src/Mod/StatString.h

```diff
@@ -103,27 +101,39 @@ or eliminating the first name as needed.  If this string is still too
 long, nothing will be changed.
  */
 
+static constexpr bool STATSTRING_GLOBALRULE_DEFAULT = false;
+
 class StatString
 {
-private:
-	std::string _stringToBeAddedIfAllConditionsAreMet;
-	std::vector<StatStringCondition*> _conditions;
-	static StatStringCondition *getCondition(const std::string &conditionName, const YAML::Node &node);
-public:
+  private:
+	/// string that is added if conditions are met.
+	std::string _conditionString;
+	std::vector<std::unique_ptr<const StatStringCondition>> _conditions;
```

Comment:
Why `std::unique_ptr`? could it be simple vector of `StatStringCondition`?

Reply:
This is the base class, but will be filled with derived, so loading them in would result in slicing. (I've disabled the copy ctor on the base to prevent this from being done).

Unless there is another way I'm not aware of?

Reply:
Ok, I did not notice that you use inherence there, probably this is overkill, one switch or function pointer could get job done but as this is code for UI and performance difference will be hard to see.

Here more detailed expatiation of this:
https://www.youtube.com/watch?v=tD5NrevFtbU
Of corse this do not mean that "Clean" code is bad, guy is bit bias there, it have simply have goal of long term maintainability, and only in permanence sensitive parts is should be abandon/avoided.

for alternatives, C++17 added `std::variant` that could solve problems like this if you know ahead of time what types will you use.